stdsimd/arch/detect/os/aarch64.rs

```diff
@@ -0,0 +1,64 @@
+//! Run-time feature detection for Aarch64 on any OS that traps the mrs instruction.
+
+use arch::detect::Feature;
+use arch::detect::cache;
+
+/// Try to read the features from the system registers.
+///
+/// This will cause SIGILL when the current OS is not trapping the mrs instruction.
+pub(crate) fn detect_features() -> cache::Initializer {
+    let mut value = cache::Initializer::default();
+
+    {
+        let mut enable_feature = |f, enable| {
+            if enable {
+                value.set(f as u32);
+            }
+        };
+
+        // ID_AA64ISAR0_EL1 - Instruction Set Attribute Register 0
+        let aa64isar0: u64;
+        unsafe { asm!("mrs $0, S3_0_C0_C6_0" : "=r"(aa64isar0)); }
+
+        let aes = bits_shift(aa64isar0, 7, 4) >= 1;
+        let pmull = bits_shift(aa64isar0, 7, 4) >= 2;
+        let sha1 = bits_shift(aa64isar0, 11, 8) >= 1;
+        let sha2 = bits_shift(aa64isar0, 15, 12) >= 1;
+        enable_feature(Feature::pmull, pmull);
+        // Crypto is specified as AES + PMULL + SHA1 + SHA2 per LLVM/hosts.cpp
+        enable_feature(Feature::crypto, aes && pmull && sha1 && sha2);
+        enable_feature(Feature::lse, bits_shift(aa64isar0, 23, 20) >= 1);
```

Comment:
According to [this page](http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.100458_0200_00_en/lau1442496521966.html) the bits here correspond to the "atomic" feature, although I'm pretty unfamiliar with AArch64 feature names so `lse` may actually be that! Just wanted to confirm

Reply:
`lse` is `atomic`, look at the comments in the `Feature` struct and the Linux implementation

Reply:
Ok cool, thanks for the info!